task_processing/runners/runner.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import abc
+
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Runner(object):
```

Comment:
I'm not sure I understand the full difference between runner and taskexecutor

Let's docstring it?

Also can we move this to a separate top level module?

e.g.

```
task_processing/
├── interface
│   ├── task_config.py
│   ├── task_executor.py
│   └── task_runner.py
└── plugins
    ├── mesos
    │   ├── mesos_executor.py
    │   └── multi_mesos_executor.py
    └── registry.py
```

This way we can use extra_requires to make it so folks can depend on the interface without depending on the mesos implementation details
```